core/textile/listener.go

```diff
@@ -87,6 +92,9 @@ func (tl *textileThreadListener) Listen(ctx context.Context) error {
 			case val, ok := <-channel:
 				if ok {
 					listenerEventHandler(val)
+				} else {
+					tl.Close()
```

Comment:
if the channel is closed, there is no point on keeping the loop alive, we must consider logic around restarting listeners if this happen